### ISSUE TITLE
spark 3.x support

### ISSUE THREAD

**Issue by antonkulaga** [polynote/polynote#926]
Spark 3.x is released ( https://spark.apache.org/releases/spark-release-3-0-0.html ), would be cool to have it in polynote

**Comment by fivejjs** [comment]
This is desirable feature.

**Comment by aaronzo** [comment]
What exactly doesn't work for you? I've just installed spark 3 and used the latest polynote-2.12.tar.gz release which uses scala 2.12, the same version that spark 3.0.0 uses, and All is running swimmingly so far.

My setup is just a local dev setup though, spark running with `--master local[*]` on WSL.

**Comment by hohonuuli** [comment]
Spark 3 in polynote 0.3.11 seems to work just fine. You can try it out in Docker:

```
docker run -p 8192:8192 -v $HOME/Documents/Notebooks:/opt/polynote/notebooks --name polynote mbari/polynote:0.3.11
```

Be sure to change the path of the volume `$HOME/Documents/Notebooks` to the path of a directory on your local computer.

**Comment by marrocksd** [comment]
![image](https://user-images.githubusercontent.com/58497107/87379864-6c9ce800-c5bb-11ea-977f-e281ad9233cf.png)

The kernel cant start if I try to add some spark properties in UI

**Comment by antonkulaga** [comment]
I am having issues like:
```
polynote.kernel.remote.SocketTransport$ProcessDied$: Kernel died unexpectedly

Uncaught exception: Kernel died unexpectedly (polynote.kernel.remote.SocketTransport$ProcessDied$)
```

**Comment by antonkulaga** [comment]
I managed to sovle kernel-died error, but now reading csv-s do not work. They run for ages without providing the results

**Comment by antonkulaga** [comment]
A simple code like:
```
val df = spark.read.format("csv").option("header", true).option("sep", "\t").load("/data/ensembl/99/ensembl_anage_vertebrates.tsv")
df.printSchema()
```
runs for ages in polynote for spark 3

**Comment by aaronzo** [comment]
As a workaround can you read the csv from a python cell or read the csv
with pandas and convert?

**Comment by antonkulaga** [comment]
I managed to make csv reading work with spark 3. Not sure how. If others do not have issues with spark 3 I am ok to close the issue

**Comment by kumarvc** [comment]
@antonkulaga 
I am also facing similar issue with spark 3.

polynote.kernel.remote.SocketTransport$ProcessDied$: Kernel died unexpectedly

Uncaught exception: Kernel died unexpectedly (polynote.kernel.remote.SocketTransport$ProcessDied$)

How it was resolved? It would be good to understand how it was resolved.

**Comment by antonkulaga** [comment]
As I am not having this issue now I will probably close

**Comment by Mehdi-Bendriss** [comment]
@kumarvc in my case it was due to the fact that I was running Spark 3.x  with the Scala 2.11 Polynote build. 
It was fixed when switching to the scala 2.12 Polynote release.
